hal/architecture/ESP8266/MyHwESP8266.cpp

```diff
@@ -29,7 +30,9 @@ bool hwInit(void)
 #endif
 #endif
 	EEPROM.begin(EEPROM_size);
-	return true;
+	// register _process() to be called at most every 1us,
+	// at every loop() or yield()
+	return schedule_recurrent_function_us([](){ _process(); return true; }, 1);
```

Comment:
Is this approach compatible with the v2 version of the Arduino core?

Reply:
Appears to be working with v2.7.4, too:

```
> Executing task in folder mysensors-arduino-test: platformio run --target upload --target monitor --environment d1_mini <

Processing d1_mini (framework: arduino; platform: espressif8266@2.6.3; board: d1_mini)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/d1_mini.html
PLATFORM: Espressif 8266 (2.6.3) > WeMos D1 R2 and mini
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 3.20704.0 (2.7.4) 
 - tool-esptool 1.413.0 (4.13) 
 - tool-esptoolpy 1.30000.201119 (3.0.0) 
 - tool-mklittlefs 1.203.210628 (2.3) 
 - tool-mkspiffs 1.200.0 (2.0) 
 - toolchain-xtensa 2.40802.200502 (4.8.2)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Library Manager: Installing git+https://github.com/d-a-v/MySensors#noCoreRedefinition
git version 2.33.1
Cloning into '/Users/kaspars/.platformio/.cache/tmp/pkg-installing-mxbmt7us'...
Library Manager: MySensors @ 2.4.0-alpha+sha.5b45fe6 has been installed!
Found 30 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <MySensors> 2.4.0-alpha+sha.5b45fe6
|   |-- <SPI> 1.0
|   |-- <Wire> 1.0
|   |-- <EEPROM> 1.0
|   |-- <ESP8266WiFi> 1.0
Building in release mode
Compiling .pio/build/d1_mini/src/main.cpp.o
Generating LD script .pio/build/d1_mini/ld/local.eagle.app.v6.common.ld
Compiling .pio/build/d1_mini/lib6fb/SPI/SPI.cpp.o
Compiling .pio/build/d1_mini/lib68b/Wire/Wire.cpp.o
Compiling .pio/build/d1_mini/libb79/EEPROM/EEPROM.cpp.o
Archiving .pio/build/d1_mini/libb79/libEEPROM.a
Indexing .pio/build/d1_mini/libb79/libEEPROM.a
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/BearSSLHelpers.cpp.o
Archiving .pio/build/d1_mini/lib6fb/libSPI.a
Indexing .pio/build/d1_mini/lib6fb/libSPI.a
Archiving .pio/build/d1_mini/lib68b/libWire.a
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/CertStoreBearSSL.cpp.o
Indexing .pio/build/d1_mini/lib68b/libWire.a
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFi.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiAP.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiGratuitous.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiSTA-WPS.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiSTA.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/ESP8266WiFiScan.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiClient.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiClientSecureAxTLS.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiClientSecureBearSSL.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiServer.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiServerSecureAxTLS.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiServerSecureBearSSL.cpp.o
Compiling .pio/build/d1_mini/libb9c/ESP8266WiFi/WiFiUdp.cpp.o
Compiling .pio/build/d1_mini/lib1bf/MySensors/MyASM.S.o
Archiving .pio/build/d1_mini/lib1bf/libMySensors.a
Indexing .pio/build/d1_mini/lib1bf/libMySensors.a
Archiving .pio/build/d1_mini/libFrameworkArduinoVariant.a
Indexing .pio/build/d1_mini/libFrameworkArduinoVariant.a
Compiling .pio/build/d1_mini/FrameworkArduino/Crypto.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Esp-frag.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Esp-version.cpp.o
Archiving .pio/build/d1_mini/libb9c/libESP8266WiFi.a
Indexing .pio/build/d1_mini/libb9c/libESP8266WiFi.a
Compiling .pio/build/d1_mini/FrameworkArduino/Esp.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/FS.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/FSnoop.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/FunctionalInterrupt.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/HardwareSerial.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/IPAddress.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/MD5Builder.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Print.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Schedule.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/StackThunk.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Stream.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/StreamString.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Tone.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/TypeConversion.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/Updater.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/WMath.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/WString.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/abi.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/base64.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/cbuf.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/cont.S.o
Compiling .pio/build/d1_mini/FrameworkArduino/cont_util.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_app_entry_noextra4k.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_eboot_command.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_features.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_flash_quirks.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_flash_utils.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_i2s.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_main.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_noniso.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_phy.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_postmortem.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_si2c.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_sigma_delta.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_spi_utils.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_timer.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_waveform.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_wiring.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_wiring_analog.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_wiring_digital.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_wiring_pulse.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_wiring_pwm.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/core_esp8266_wiring_shift.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/crc32.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/debug.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/flash_hal.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/gdb_hooks.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/heap.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/libb64/cdecode.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/libb64/cencode.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/libc_replacements.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/sntp-lwip2.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/spiffs/spiffs_cache.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/spiffs/spiffs_check.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/spiffs/spiffs_gc.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/spiffs/spiffs_hydrogen.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/spiffs/spiffs_nucleus.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/spiffs_api.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/sqrt32.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/time.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/uart.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/umm_malloc/umm_info.c.o
Compiling .pio/build/d1_mini/FrameworkArduino/umm_malloc/umm_integrity.c.o
Compiling .pio/build/d1_mini/FrameworkArduino/umm_malloc/umm_local.c.o
Compiling .pio/build/d1_mini/FrameworkArduino/umm_malloc/umm_malloc.cpp.o
Compiling .pio/build/d1_mini/FrameworkArduino/umm_malloc/umm_poison.c.o
Archiving .pio/build/d1_mini/libFrameworkArduino.a
Indexing .pio/build/d1_mini/libFrameworkArduino.a
Linking .pio/build/d1_mini/firmware.elf
Building .pio/build/d1_mini/firmware.bin
Retrieving maximum program size .pio/build/d1_mini/firmware.elf
Checking size .pio/build/d1_mini/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [===       ]  33.8% (used 27716 bytes from 81920 bytes)
Flash: [===       ]  26.1% (used 272864 bytes from 1044464 bytes)
Creating BIN file ".pio/build/d1_mini/firmware.bin" using "/Users/kaspars/.platformio/packages/framework-arduinoespressif8266@3.20704.0/bootloaders/eboot/eboot.elf" and ".pio/build/d1_mini/firmware.elf"
Configuring upload protocol...
AVAILABLE: espota, esptool
CURRENT: upload_protocol = esptool
Looking for upload port...
Auto-detected: /dev/cu.usbserial-1420
Uploading .pio/build/d1_mini/firmware.bin
esptool.py v3.0
Serial port /dev/cu.usbserial-1420
Connecting....
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: e8:db:84:dc:42:fc
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Compressed 277024 bytes to 203728...
Writing at 0x00000000... (7 %)
Writing at 0x00004000... (15 %)
Writing at 0x00008000... (23 %)
Writing at 0x0000c000... (30 %)
Writing at 0x00010000... (38 %)
Writing at 0x00014000... (46 %)
Writing at 0x00018000... (53 %)
Writing at 0x0001c000... (61 %)
Writing at 0x00020000... (69 %)
Writing at 0x00024000... (76 %)
Writing at 0x00028000... (84 %)
Writing at 0x0002c000... (92 %)
Writing at 0x00030000... (100 %)
Wrote 277024 bytes (203728 compressed) at 0x00000000 in 19.8 seconds (effective 112.1 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
========================================================================== [SUCCESS] Took 45.64 seconds ==========================================================================
--- Available filters and text transformations: colorize, debug, default, direct, esp8266_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Miniterm on /dev/cu.usbserial-1420  9600,8,N,1 ---
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
5096 !MCO:SND:NODE NOT REG
10124 !MCO:SND:NODE NOT REG
15154 !MCO:SND:NODE NOT REG
```

Reply:
Thanks for testing/reviewing !

> Is this approach compatible with the v2 version of the Arduino core?

It should work with v2.6+ (v2.6.x, v2.7.x, v3.0.x) of the esp8266 arduino core.
This API was not present in v2.5.x.